### PR TITLE
Integrate cosign - an implementation of the update framework in ballerina dependency releases

### LINKED ISSUE #34: The update framework (TUF) integration
**Description:**
Integrating a component of The Update Framework (TUF), this can be done via `Sigstore's Cosign` Tool. `Cosign` can be used to sign and verify the release artifacts, in this case would be to sign and release the Ballerina Dependencies. This would further provide user's the confidence that `Ballerina` is improving the security in its supply chain.

### PR DESCRIPTION
**Pull request by Miranlfk**
## Purpose
$title

## Goals
Fixes #34

### CODE REVIEW

File: .github/workflows/publish-mac-arm.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        env:
+          COSIGN_EXPERIMENTAL: "true"
+        run: |
+          cosign sign-blob ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.zip --output-certificate ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.pem --output-signature ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.sig --yes
+      - name: Verify the MacOS-ARM Artifact
+        env:
+          COSIGN_EXPERIMENTAL: "true"
+        run: |
+          cosign verify-blob ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.zip --certificate ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.pem --signature ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.sig --certificate-identity=https://github.com/${{ github.actor }}/ballerina-custom-jre/.github/workflows/publish-mac-arm.yml@refs/heads/main --certificate-oidc-issuer=https://token.actions.githubusercontent.com 
+      - uses: IsaacShelton/update-existing-release@v1.3.2
```
Comment (keizer619) [inline .github/workflows/publish-mac-arm.yml]: @Miranlfk Shall we use an official action like ` actions/upload-release-asset`

Comment (Miranlfk) [reply .github/workflows/publish-mac-arm.yml:50]: I'll make the change

Comment (Miranlfk) [reply .github/workflows/publish-mac-arm.yml]: The official action cannot be used in the case of updating an existing release, as per my testing. I've opted to use `github-cli` to do the functionality

File: .github/workflows/publish-mac-arm.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    if: github.repository_owner == 'ballerina-platform'
+    steps:
+      - id: checkout
+        uses: actions/checkout@v2
+      - id: execute
+        run: |
+          export VERSION="$((grep -w 'version' | cut -d= -f2) < gradle.properties | cut -d- -f1)"
+          echo "::set-output name=version::${VERSION}"
+    outputs:
+      version: ${{steps.execute.outputs.version}}
```
Comment (keizer619) [inline .github/workflows/publish-mac-arm.yml]: @Miranlfk Shall we include version as an user input?

Comment (Miranlfk) [reply .github/workflows/publish-mac-arm.yml]: This has been completed

**Review by keizer619 — APPROVED** [review]

**Pull request merged**